contracts/HATArbitrator.sol

```diff
@@ -0,0 +1,200 @@
+    error CannotClaimBond();
+
+    using SafeERC20 for IERC20;
+
+    struct Resolution {
+        address beneficiary;
+        uint16 bountyPercentage;
+        uint256 resolvedAt;
+    }
+
+    IHATVault public vault;
+    address public expertCommittee;
+    address public court;
+    IERC20 public token;
+    uint256 public bondsNeededToStartDispute;
+    uint256 public minBondAmount;
+
+    mapping(address => mapping(bytes32 => uint256)) public disputersBonds;
+    mapping(address => mapping(bytes32 => bool)) public bondClaimable;
+    mapping(bytes32 => uint256) public totalBondsOnClaim;
+    mapping(bytes32 => Resolution) public resolutions;
+    mapping(bytes32 => uint256) public resolutionChallengedAt;
+
+    event ClaimDisputed(bytes32 indexed _claimId, address indexed _disputer, bytes32 _ipfsHash, uint256 _bondAmount);
+
+    modifier onlyExpertCommittee() {
+        if (msg.sender != expertCommittee) {
+            revert OnlyExpertCommittee();
+        }
+        _;
+    }
+
+    modifier onlyChallengedActiveClaim(bytes32 _claimId) {
+        (bytes32 claimId,,,,,uint32 challengedAt,,,,,,,) = vault.activeClaim();
+
+        if (claimId != _claimId) {
+            revert ClaimDisputedIsNotCurrentlyActiveClaim();
+        }
+
+        if (challengedAt == 0) {
+            revert ClaimIsNotDisputed();
+        }
+        _;
+    }
+
+    modifier onlyUnresolvedDispute(bytes32 _claimId) {
+        if (resolutions[_claimId].resolvedAt != 0) {
+            revert AlreadyResolved();
+        }
+        _;
+    }
+
+    modifier onlyResolvedDispute(bytes32 _claimId) {
+        if (resolutions[_claimId].resolvedAt == 0) {
+            revert NoResolution();
+        }
+        _;
+    }
+
+    constructor (IHATVault _vault, address _expertCommittee, address _court, IERC20 _token, uint256 _bondsNeededToStartDispute, uint256 _minBondAmount) {
+        vault = _vault;
+        expertCommittee = _expertCommittee;
+        court = _court;
+        token = _token;
+        bondsNeededToStartDispute = _bondsNeededToStartDispute;
+        minBondAmount = _minBondAmount;
+        if (minBondAmount > bondsNeededToStartDispute) {
+            revert bondsNeededToStartDisputeMustBeHigherThanMinAmount();
+        }
+    }
+
+    function dispute(bytes32 _claimId, bytes32 _ipfsHash, uint256 _bondAmount) external {
+        if (_bondAmount < minBondAmount) {
+            revert BondAmountSubmittedTooLow();
+        }
+
+        (bytes32 claimId,,,,,uint32 challengedAt,,,,,,,) = vault.activeClaim();
+        if (claimId != _claimId) {
+            revert ClaimDisputedIsNotCurrentlyActiveClaim();
+        }
+
+        disputersBonds[msg.sender][_claimId] += _bondAmount;
+        totalBondsOnClaim[_claimId] += _bondAmount;
+
+        token.safeTransferFrom(msg.sender, address(this), _bondAmount);
+
+        if (totalBondsOnClaim[_claimId] >= bondsNeededToStartDispute) {
+            if (challengedAt == 0) {
+                vault.challengeClaim(_claimId);
```

Comment:
but it would be nicer if it did not require an explciit call by the expert committeel

Reply:
What if they want to only refund some? (Ie. some were spam)

Reply:
well, in this case, there was not enough money collected during the challengePeriod to actually start a dispute. So the original claim will be executed. So even if it  was spam, it did not do any harm. And there is nobody being paid to distinguish spam from legit claims. So yeah, I think all disputers should be refunded automatically here, instead of it requiring an action by the expert committee (which will then not be paid for that)